pkgs/development/python-modules/pycuda/default.nix

```diff
@@ -29,11 +29,12 @@ buildPythonPackage rec {
     sha256 = "ada56ce98a41f9f95fe18809f38afbae473a5c62d346cfa126a2d5477f24cc8a";
   };
 
-  preConfigure = ''
+  preConfigure = with stdenv.lib.versions; ''
     ${python.interpreter} configure.py --boost-inc-dir=${boost.dev}/include \
                           --boost-lib-dir=${boost}/lib \
                           --no-use-shipped-boost \
-                          --boost-python-libname=boost_python${stdenv.lib.optionalString isPy3k "3"}
+                          --boost-python-libname=boost_python${major python.version}${minor python.version} \
```

Comment:
I think you can try to drop this line. From my understanding it should still compile.

Reply:
It fails with
```
y37 -lboost_thread -lcuda -lcurand -lpython3.7m -o build/lib.linux-x86_64-3.7/pycuda/_driver.cpython-37m-x86_64-linux-gnu.so
/nix/store/a57856fs4m8ir6vlv14h3gq3sv9aq2lb-binutils-2.31.1/bin/ld: cannot find -lboost_python-py37
collect2: error: ld returned 1 exit status
```

Reply:
note the presence of the dash in the library name which shouldn't be there

Reply:
makes sense.